connect/evidence/containers.py

```diff
@@ -61,6 +62,15 @@ def _validate_inputs(self, data):
     def _validate(self, data):
         pass
 
+    def remove_NaNs(self):
```

Comment:
I think this may need to be more robust. What happens when this is a dictionary? 

Also, the way you use it (always calling it in `to_evidence`) you don't need to change `_data` in place. Just have it do the transformation on the data, and return the cleaned data for export.

Reply:
Still working on the data profiler sanitization. It is complicated because it contains nested dictionaries and then some DataFrames/Series within those dictionaries. It's _probably_ safe to assume those DataFrames aren't further nested (i.e. they just contain elementary types) but if we don't want to assume that we'll need a very complicated sanitizer.

On further thought, I changed the paradigm of the call. This should be a forced sanitization so I've moved the call to the base class `EvidenceContainer`'s init function. The function is now abstract and forces subclasses to implement it. This will help prevent some future developer from implementing a new evidence while forgetting to sanitize for JSONs.

I don't really see the harm in having self._data reflect a de-NaNified structure. As currently written, Evidence _always_ gets converted to a JSON. The prior implementation isn't really "in place" --> I was passing the _data object on the RHS and the return was a copy assigned to the same variable name. The way I'm doing it now precludes that by just sanitizing at the start (sanitizing a copy so we don't have to worry about deep copy issues).